### PR TITLE
Creating an example for an app with backend component

### PR DESCRIPTION
**Pull request by tab0r**
The [plugin-tools](https://github.com/grafana/plugin-tools) command `yarn create @grafana/plugin` allows creation of an app with backend, but we don't have an example for it yet. Since it's what I need for a current project, I thought I'd share what I came up with.

### CODE REVIEW

**Comment by tab0r** [comment]
Creating this as a draft, since there are a few things I'd like to do/know:

- [ ] do we really want all the build files included in the apps?
- [ ] add something in the UI to display the button responses
- [ ] add something to hit the `/echo` route
- [ ] update docs
- [ ] add some integration tests
- [ ] add unit tests

**Review by mckn — COMMENTED** [review]
This is great, thanks for your contribution! 

To make this even better we should probably do the following:

- Replace some of the pages with one or two pages that show case this example a bit more by calling the app backend and displaying the response.
- Add some e2e tests that verifies that the page that calls the app backend works as expected.

By doing the changed mentioned above this example will also verify that we are not breaking any of the APIs being used while working on future versions of Grafana. All of the examples that contains e2e tests will regularly run agains the main branch of Grafana.

**Comment by tab0r** [comment]
I definitely agree with both points @mckn . I'm not sure how much of that I can get done in the short term, as I'm honestly not sure how much work it entails. But, tomorrow I'll be pairing with a teammate to discuss our implementation plans. There's a chance that taking care of those things will be just want we need to get started on our project. If not, I wouldn't be opposed to closing this and creating an issue for it so that it can be tackled as availability opens up.

**Comment by mckn** [comment]
> I definitely agree with both points @mckn . I'm not sure how much of that I can get done in the short term, as I'm honestly not sure how much work it entails. But, tomorrow I'll be pairing with a teammate to discuss our implementation plans. There's a chance that taking care of those things will be just want we need to get started on our project. If not, I wouldn't be opposed to closing this and creating an issue for it so that it can be tackled as availability opens up.

Okay, no worries! I can help you with those changes if that feels ok? @wbrowne before I proceed, can you have a look at the BE code?

File: examples/app-with-backend/src/components/AppConfig/AppConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  isApiKeySet?: boolean;
+};
+
+type State = {
+  // The URL to reach our custom API.
+  apiUrl: string;
+  // Tells us if the API key secret is set.
+  // Set to `true` ONLY if it has already been set and haven't been changed.
+  // (We unfortunately need an auxiliray variable for this, as `secureJsonData` is never exposed to the browser after it is set)
+  isApiKeySet: boolean;
```
Comment (wbrowne) [inline examples/app-with-backend/src/components/AppConfig/AppConfig.tsx:20]: Here we can make use of `secureJsonFields KeyValue<boolean>` to achieve this behaviour without the added field. This should exist on the PluginMeta. 

For reference the PR that added support: https://github.com/grafana/grafana/pull/55313

**Review by wbrowne — CHANGES_REQUESTED** [review]
Just a couple of small things that should be pretty easily fixed 😄 Thanks for providing this!

**Comment by tab0r** [comment]
Thank you @mckn and @wbrowne ! After the team discussion I mentioned we decided to not go down this path at the moment. But now I'm invested! So still keeping an eye on this and will jump back in when I can. :smile:

**Review by mckn — APPROVED** [review]
LGTM!

File: examples/app-with-backend/src/components/AppConfig/AppConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Button, useStyles2, FieldSet } from '@grafana/ui';
+import { PluginConfigPageProps, AppPluginMeta, PluginMeta, GrafanaTheme2 } from '@grafana/data';
+import { getBackendSrv, locationService } from '@grafana/runtime';
+import { css } from '@emotion/css';
+import { testIds } from '../testIds';
+
+export type JsonData = {
+  apiUrl?: string;
+  isApiKeySet?: boolean;
```
Comment (wbrowne) [inline examples/app-with-backend/src/components/AppConfig/AppConfig.tsx:10]: This can be removed now I think!

**Review by wbrowne — APPROVED** [review]
LGTM.

Spoke offline and we will move to `secureJsonFields` support once G10 rolls out

**Comment by tab0r** [comment]
Thank you so much for cleaning this up for me @mckn ! I'll leave the pleasure of pushing the big green button to you, it's only fair. :D

**Pull request merged**